internal/dao/sqlx/account_dao.go

```diff
@@ -113,7 +111,7 @@ func (di *accountDaoSqlx) GetOrCreateByIdentity(ctx context.Context, orgId strin
 	} // else it is NoRows and we just continue
 	ctxval.Logger(ctx).Debug().Msgf("Account not found by account number %s", accountNumber)
 
-	acc = &models.Account{OrgID: orgId, AccountNumber: ptr.String(accountNumber)}
+	acc = &models.Account{OrgID: orgId, AccountNumber: sql.NullString{String: accountNumber, Valid: true}}
```

Comment:
Here this code needs to actually make use of NULL:

* When accountNumber is blank (https://github.com/RedHatInsights/identity/blob/main/identity.go#L48 - it is not a pointer so the only option is blank) it needs to set Valid to false.
* Otherwise it needs to create new struct with Valid = true.